feat(cli): promote tasks commands from experimental to GA

mafredri opened the pull request:
## Overview

This PR promotes the tasks CLI commands from `coder exp task` to `coder task`, marking them as generally available (GA).

## Changes

- **CLI Structure**: Moved tasks commands from experimental subcommand to core commands
- **File Renames**: Renamed all `exp_task*.go` files to `task*.go`
- **Command Registration**: Updated `root.go` to register tasks in `CoreSubcommands()` instead of `AGPLExperimental()`
- **Documentation**: Updated command descriptions and examples to remove 'experimental' references
- **Tests**: Updated all test files to use the new command path
- **Golden Files**: Regenerated CLI help output golden files
- **Documentation**: Updated `docs/ai-coder/cli.md` with the new command paths and removed experimental language

## Migration

Users will need to update their scripts from:
```bash
coder exp task create "my task"
```

To:
```bash
coder task create "my task"
```

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.

mafredri committed:
feat(cli): promote tasks commands from experimental to GA

Move tasks CLI commands from 'coder exp task' to 'coder task' to mark them
as generally available. This change includes:

- Rename all task-related files from exp_task*.go to task*.go
- Move tasksCommand() from AGPLExperimental() to CoreSubcommands() in root.go
- Update command descriptions to remove 'experimental' references
- Update all test files to use 'coder task' instead of 'coder exp task'
- Update CLI examples in all command files
- Update documentation in docs/ai-coder/cli.md
- Regenerate golden files for new command structure

The tasks feature is now stable and ready for production use.

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.

mafredri committed:
chore(docs): add generated CLI reference docs for task commands

Add auto-generated CLI reference documentation that was missed in the initial
commit. These files are created by 'make gen' and include:

- docs/reference/cli/task*.md files
- Updated docs/manifest.json
- Updated docs/reference/cli/index.md

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.

mafredri committed:
fix(cli): update task_test.go to use new command path

Update Test_Tasks to use 'task' instead of 'exp task' in command
invocations to match the new command structure.

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.

DanielleMaywood reviewed (approved):
Approved but agree with removal of file

mafredri committed:
docs: simplify ai-coder/cli.md with links to generated docs

Replace the detailed CLI documentation with a simple redirect page that
links to the auto-generated CLI reference documentation. This reduces
duplication and ensures users always see the most up-to-date CLI docs.

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.

mafredri committed:
Merge remote-tracking branch 'origin/main' into promote-tasks-cli-to-ga

mafredri committed:
chore: update generated docs after merge with main

Regenerate CLI docs and golden files to reflect the new task display_name
field that was added in main branch.

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏄🏻‍♂️.
